BLUESPAWN-client/src/hunt/hunts/HuntT1122.cpp

```diff
@@ -0,0 +1,70 @@
+#include "hunt/hunts/HuntT1122.h"
+#include "hunt/RegistryHunt.h"
+
+#include "util/configurations/Registry.h"
+#include "util/filesystem/Filesystem.h"
+#include "util/log/Log.h"
+#include "util/log/HuntLogMessage.h"
+
+#include "common/Utils.h"
+
+#include <algorithm>
+
+using namespace Registry;
+
+namespace Hunts{
+
+	HuntT1122::HuntT1122() : Hunt(L"T1122 - COM Hijacking"){
+		dwSupportedScans = (DWORD) Aggressiveness::Intensive;
+		dwCategoriesAffected = (DWORD) Category::Configurations;
+		dwSourcesInvolved = (DWORD) DataSource::Registry;
+		dwTacticsUsed = (DWORD) Tactic::Persistence;
+	}
+
+	int HuntT1122::ScanIntensive(const Scope& scope, Reaction reaction){
+		LOG_INFO("Hunting for " << name << " at level Cursory");
+		reaction.BeginHunt(GET_INFO());
+
+		int detections = 0;
+
+		std::map<std::wstring, std::vector<RegistryKey>> files{};
+
+		for(auto key : CheckSubkeys(HKEY_LOCAL_MACHINE, L"SOFTWARE\\Classes\\CLSID", true, true)){
```

Comment:
Would it be better to just enumerate everything in HKEY_CLASSES_ROOT\CLSID instead?

Reply:
HKEY_CLASSES_ROOT is a symlink to KEY_LOCAL_MACHINE\Software\Classes. I also want to scan users (and if a WoW64 redirected key exists, that too)